components/forms/TaskForm.tsx

```diff
@@ -0,0 +1,125 @@
+import { Checkbox, DatePicker, Form, Input, InputNumber, Select, TimePicker } from 'antd';
+import { CSSProperties, Dispatch, SetStateAction, useEffect, useState } from 'react';
+import { NotificationDateTimeType, TaskFormData } from '../../types/task';
+import { Nullable } from '../../types/utils';
+
+interface Props {
+  initialValues: TaskFormData;
+  shouldShowMarkDone: boolean;
+  isSubmitting: boolean;
+  setTaskFormData: Dispatch<SetStateAction<Nullable<TaskFormData>>>;
+}
+
+interface TaskFormTimePickerProps {
+  isOpen: boolean;
+  givenStyle?: CSSProperties;
+}
+
+const { Option } = Select;
+
+function TaskForm({ initialValues, shouldShowMarkDone, isSubmitting, setTaskFormData }: Props) {
+  const [form] = Form.useForm();
+
+  const [shouldShowNotificationDaysOffsetInput, setShouldShowNotificationDaysOffsetInput] = useState<boolean>(false);
+  const [shouldShowNotificationDateTimePicker, setShouldShowNotificationDateTimePicker] = useState<boolean>(false);
+  const [shouldShowNotificationTimePicker, setShouldShowNotificationTimePicker] = useState<boolean>(false);
+
+  useEffect(() => {
+    form.resetFields();
+    onSelectNotificationDateTimeType(initialValues.notificationDateTimeType ?? NotificationDateTimeType.NONE);
+  }, [initialValues]);
+
+  useEffect(() => {
+    if (isSubmitting) {
+      setTaskFormData(form.getFieldsValue());
+    }
+  }, [isSubmitting]);
```

Comment:
Validation does not work.

```suggestion
  useEffect(() => {
    if (isSubmitting) {
      form.validateFields().then(() => setTaskFormData(form.getFieldsValue()));
    }
  }, [isSubmitting]);
```